### DISCUSSION TITLE
Problems with setfuse

### DISCUSSION THREAD

**Discussion by 6tus** [dfu-programmer/dfu-programmer#99] [Q&A]
Hi.

I have a board I'm trying to flash with dfu-programmer. The project requires changing the fuses and that's where I ran into a problem.

I have a board for testing, which was flashed with Flip and it has correct fuse settings. I'm currently trying to replicate the flashing with dfu-programmer.

![image](https://github.com/dfu-programmer/dfu-programmer/assets/34712938/fc2b753f-b1dc-4a4b-9589-0b424f1eddc3)

When I first read the LOCK fuses, it gives the correct value 0xFFFF. But when I try to write 0xFFFF with dfu-programmer it writes just 0xFF. 

I also ran into an issue with BODLEVEL fuses. When I'm trying to write 0x27, dfu-programmer writes 0x17. Tested this with few values and it's always the same. 0x25->0x15, 0x0F->0x1F etc.

Some kind of help would be appreciated.

**Comment by cinderblock** [comment]
This looks like a bug to me. Created an issue #100 to track it.

Working on setting up a test host so we have some repeatable hardware to test against. However life is keeping this from being a priority right now.
